src/pyspiffe/svid/jwt_svid_validator.py

```diff
@@ -0,0 +1,132 @@
+"""
+This module manages the validations of JWT tokens.
+"""
+
+import datetime
+from typing import List, Dict
+from calendar import timegm
+
+from pyspiffe.svid import INVALID_INPUT_ERROR
+
+from pyspiffe.svid.exceptions import (
+    TokenExpiredError,
+    InvalidClaimError,
+    InvalidAlgorithmError,
+    InvalidTypeError,
+)
+
+AUDIENCE_NOT_MATCH_ERROR = 'audience does not match expected value'
+"""str: audience does not match error message."""
+
+
+class JwtSvidValidator(object):
+    """Performs validations on a given token checking compliance to SPIFFE specification.
+    See `SPIFFE JWT-SVID standard <https://github.com/spiffe/spiffe/blob/master/standards/JWT-SVID.md>`
+
+    """
+
+    _REQUIRED_CLAIMS = ['aud', 'exp', 'sub']
+    _SUPPORTED_ALGORITHMS = [
+        'RS256',
+        'RS384',
+        'RS512',
+        'ES256',
+        'ES384',
+        'ES512',
+        'PS256',
+        'PS384',
+        'PS512',
+    ]
+    _SUPPORTED_TYPES = ['JWT', 'JOSE']
+
+    def __init__(self) -> None:
+        pass
+
+    def validate_header(self, header: Dict) -> None:
+        """Validates token header by verifing if header specifies supported algortihms and token type. Type is optional but in case it is present, it must be set to the supported values.
+
+        Args:
+            header: token header.
+
+        Returns:
+            None.
+
+        Raises:
+            ValueError: in case header is not specified.
+            InvalidAlgorithmError: in case specified 'alg' is not supported as specified by the SPIFFE standard.
+            InvalidTypeError: in case 'typ' is present in header but is not set to 'JWT' or 'JOSE'.
+        """
+        if not header:
+            raise ValueError(INVALID_INPUT_ERROR.format('header alg cannot be empty'))
+
+        alg = header.get('alg')
+        if not alg:
+            raise ValueError(INVALID_INPUT_ERROR.format('header alg cannot be empty'))
+
+        if alg not in self._SUPPORTED_ALGORITHMS:
+            raise InvalidAlgorithmError(alg)
+
+        typ = header.get('typ')
+        if typ and typ not in self._SUPPORTED_TYPES:
+            raise InvalidTypeError(typ)
+
+    def validate_claims(self, payload: Dict, expected_audience: List) -> None:
```

Comment:
nit: `audience: List[str]` For `payload: Dict` you can also define the dict structure if it is fixed like `Dict[str, str]` for example.

Reply:
The payload is not fixed so, it does not apply. But header is, so, I've changed for header.